src/components/slider/ColorSliderGroup.tsx

```diff
@@ -42,79 +42,82 @@ export type ColorSliderGroupProps = {
    * If true the component will have accessibility features enabled
    */
   accessible?: boolean;
-  /** 
+  /**
    * Whether to use the new Slider implementation using Reanimated
    */
   migrate?: boolean;
 };
 
-interface ColorSliderGroupState {
-  initialColor: string;
-}
+type ColorSliderProps = Pick<
+  ColorSliderGroupProps,
+  'sliderContainerStyle' | 'showLabels' | 'labelsStyle' | 'accessible' | 'labels' | 'migrate' | 'initialColor'
+> & {
+  type: GradientSliderTypes;
+};
+
+const ColorSlider = (props: ColorSliderProps) => {
+  const {type, sliderContainerStyle, showLabels, labelsStyle, accessible, labels, migrate, initialColor} = props;
+  return (
+    <>
+      {showLabels && labels && (
+        <Text recorderTag={'unmask'} $textNeutral text80 style={labelsStyle} accessible={accessible}>
+          {labels[type]}
+        </Text>
+      )}
+      <GradientSlider
+        color={initialColor}
+        type={type}
+        containerStyle={sliderContainerStyle}
+        accessible={accessible}
+        migrate={migrate}
+      />
+    </>
+  );
+};
 
 /**
  * @description: A Gradient Slider component
  * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/SliderScreen.tsx
  * @gif: https://github.com/wix/react-native-ui-lib/blob/master/demo/showcase/ColorSliderGroup/ColorSliderGroup.gif?raw=true
  */
-class ColorSliderGroup extends PureComponent<ColorSliderGroupProps, ColorSliderGroupState> {
-  static displayName = 'ColorSliderGroup';
-
-  static defaultProps = {
-    labels: {hue: 'Hue', lightness: 'Lightness', saturation: 'Saturation'}
-  };
+const ColorSliderGroup = (props: ColorSliderGroupProps) => {
+  const {
+    labels = {hue: 'Hue', lightness: 'Lightness', saturation: 'Saturation', default: ''},
+    initialColor,
+    containerStyle,
+    showLabels,
+    labelsStyle,
+    accessible,
+    migrate,
+    sliderContainerStyle
+  } = props;
+  const [color, setColor] = useState(initialColor);
+  useEffect(() => {
+    setColor(initialColor);
+  }, [initialColor]);
 
-  state = {
-    initialColor: this.props.initialColor
+  const onValueChange = (value: string) => {
+    _.invoke(props, 'onValueChange', value);
   };
 
-  static getDerivedStateFromProps: GetDerivedStateFromProps<ColorSliderGroupProps, ColorSliderGroupState> = (nextProps,
-    prevState) => {
-    if (prevState.initialColor !== nextProps.initialColor) {
-      return {
-        initialColor: nextProps.initialColor
-      };
-    }
-    return null;
+  const sliderProps = {
+    initialColor,
+    sliderContainerStyle,
+    showLabels,
+    labelsStyle,
+    accessible,
+    labels,
+    migrate
```

Comment:
Seems like you could pick up the props you need for the group (container) and pass the others as "others" to the ColorSlider, no?

Reply:
Most of them. Yeah. Fixed.